orttraining/orttraining/core/framework/gradient_graph_builder.h

```diff
@@ -77,6 +77,13 @@ static std::unordered_map<std::string, std::unordered_set<size_t>>
 
 static std::unordered_set<std::string> INVERTIBLE_OPS{"LayerNormalization",
                                                       "Relu"};
+static std::unordered_set<int64_t> CAST_GRAD_ALLOWED_TYPES{
+    ONNX_NAMESPACE::TensorProto_DataType_FLOAT,
+    ONNX_NAMESPACE::TensorProto_DataType_FLOAT16,
+    ONNX_NAMESPACE::TensorProto_DataType_DOUBLE,
+    ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16,
+};
+static const std::unordered_set<size_t> CAST_STOP_EDGE{0};
```

Comment:
how is this different than "STOP_GRADIENT_EDGES" above? Why not use "STOP_GRADIENT_EDGES" instead?

Reply:
For cast, we only stop edge in specific conditions, whereas the STOP_GRADIENT_EDGES is used for always stopping the edges. The STOP_GRADITNT_EDGES is still used in the older OrtTrainer code, hence separating the definitions